pkg/shardddl/pessimism/keeper.go

```diff
@@ -18,24 +18,56 @@ import (
 	"sync"
 
 	"github.com/pingcap/tidb-tools/pkg/dbutil"
+
+	"github.com/pingcap/dm/pkg/shardddl/optimism"
+)
+
+// LockMode represents the mode of a DDL lock.
+type LockMode uint32
+
+const (
+	// LockModePessimistic sets the lock mode to be pessimistic. The lock is
+	// held until all sources have executed the same DDL upstream.
+	LockModePessimistic LockMode = iota
+	// LockModeOptimistic sets the lock mode to be optimistic. The lock is
+	// released as soon as the downstream table schema is compatible with all
+	// upstream schemas.
+	LockModeOptimistic
 )
 
 // LockKeeper used to keep and handle DDL lock conveniently.
 // The lock information do not need to be persistent, and can be re-constructed from the shard DDL info.
 type LockKeeper struct {
 	mu    sync.RWMutex
 	locks map[string]*Lock // lockID -> Lock
+	mode  LockMode
```

Comment:
Do we need to set a different `mode` for each migration task?

If so, maybe we can add a config item in task.yaml?